Update focus style to GOVUK Frontend 3 version

tombye opened the pull request:
Updates our focus style to GOVUK's new one, to make the contrast compliant.

https://github.com/alphagov/notifications-admin/pull/3303/files

This is a draft proposal, intended to show where this work is so it can have some design work.

This includes:
1. changing the core focus style for links and buttons
2. updating all the variants of links and buttons custom to Notify

Number 2. includes an attempt to update the 'pill' items which probably needs some more work.

## New GOVUK Focus style

The new focus style is explained in [this blog post](https://designnotes.blog.gov.uk/2019/07/29/weve-made-the-gov-uk-design-system-more-accessible/) and [this page on their docs](https://design-system.service.gov.uk/get-started/focus-states/).

The main thing to bear in mind, for any variants we introduce is:
- design system's design actually includes 2 styles (the colour and the underline) so part of it will always contrast with the background colour
- they need to comply with [WCAG 1.4.11](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html)

This [WCAG techniques article](https://www.w3.org/WAI/WCAG21/Techniques/css/C40) contains some useful tests for compliance of any similar design.

## Focus styles custom to Notify

Here's some screenshots of my attempt at updating the various links/buttons custom to Notify. I based this work on those found in https://github.com/alphagov/notifications-admin/pull/3303 so there's a chance I might have missed some...

### Link in dashboard table

<img width="750" alt="image" src="https://user-images.githubusercontent.com/87140/93191867-efb5f800-f73c-11ea-9f22-18bf7140adfa.png">

### Big numbers

<img width="243" alt="image" src="https://user-images.githubusercontent.com/87140/93191916-fe041400-f73c-11ea-9c9d-69ee06d898dc.png">

### Show more component

<img width="737" alt="image" src="https://user-images.githubusercontent.com/87140/93191943-08261280-f73d-11ea-9f52-864a4fe5f293.png">

### Service navigation links

<img width="205" alt="image" src="https://user-images.githubusercontent.com/87140/93192007-1b38e280-f73d-11ea-8422-1de6d44b8656.png">

### Org' navigation links

<img width="994" alt="image" src="https://user-images.githubusercontent.com/87140/93192066-2d1a8580-f73d-11ea-9db4-67196887fdd1.png">

### Previous/next navigation

<img width="415" alt="image" src="https://user-images.githubusercontent.com/87140/93192164-44f20980-f73d-11ea-9344-a32c4724c91b.png">

### Link in table cell

<img width="432" alt="image" src="https://user-images.githubusercontent.com/87140/93192241-5c30f700-f73d-11ea-9f65-2e1863216eba.png">

### Pill items

<img width="740" alt="image" src="https://user-images.githubusercontent.com/87140/93192295-6c48d680-f73d-11ea-95cc-a3ae32ab074c.png">

### Separate pill items

<img width="748" alt="image" src="https://user-images.githubusercontent.com/87140/93192334-779c0200-f73d-11ea-8e6d-10096aee2405.png">

quis commented:
### Link in dashboard table

<img width="750" alt="image" src="https://user-images.githubusercontent.com/87140/93191867-efb5f800-f73c-11ea-9f22-18bf7140adfa.png">

Think this looks crisper when the black border overlaps the grey keylines:
![image](https://user-images.githubusercontent.com/355079/93196388-46720080-f742-11ea-80f4-d29253face0f.png)

### Big numbers

<img width="243" alt="image" src="https://user-images.githubusercontent.com/87140/93191916-fe041400-f73c-11ea-9c9d-69ee06d898dc.png">

This would look better by insetting the black border so the white space above the green box is preserved (ignore the chamfer):

![image](https://user-images.githubusercontent.com/355079/93196977-fcd5e580-f742-11ea-811d-d8dcd66d6e10.png)

### Service navigation links and org navigation links

<img width="994" alt="image" src="https://user-images.githubusercontent.com/87140/93192066-2d1a8580-f73d-11ea-9db4-67196887fdd1.png">

Again, crisper if the black overlaps the grey keyline:
![image](https://user-images.githubusercontent.com/355079/93197381-766dd380-f743-11ea-9d00-4997709a84a1.png)

tombye commented:
@quis I've pushed changes for most of the things you mentioned here and on slack.

## Pill items

<img width="740" alt="image" src="https://user-images.githubusercontent.com/87140/93209991-cdc96f00-f756-11ea-922e-aa5dc75f3f13.png">

<img width="750" alt="image" src="https://user-images.githubusercontent.com/87140/93210054-e2a60280-f756-11ea-8452-4fcee2a809ee.png">

## Org nav links

<img width="991" alt="image" src="https://user-images.githubusercontent.com/87140/93210097-f3567880-f756-11ea-86f0-92826b1221ed.png">

## Link in dashboard table

<img width="637" alt="image" src="https://user-images.githubusercontent.com/87140/93210196-154ffb00-f757-11ea-9bd2-5461cc8c6530.png">

## Big numbers

<img width="256" alt="image" src="https://user-images.githubusercontent.com/87140/93210223-23058080-f757-11ea-85b0-62943df9bc3d.png">

quis commented:
A few more things I’ve noticed…

## Org nav links

<img width="991" alt="image" src="https://user-images.githubusercontent.com/87140/93210097-f3567880-f756-11ea-86f0-92826b1221ed.png">

The org link has a `1px` thicker border than switch service 

# Template page

![image](https://user-images.githubusercontent.com/355079/93215749-c27a4180-f75e-11ea-8f14-45c07b0e3a7a.png)

The focus state is being triggered on hover here

# Dashboard

![image](https://user-images.githubusercontent.com/355079/93215878-e6d61e00-f75e-11ea-8d13-7dd75b4ca8b8.png)

The blue boxes have lost their hover state (click ‘0 emails’ and look at the sending/delivered/failed tabs for an example of what the hover should look like)

The focus state is triggered on hover

quis commented:
Also `placeholders.scss` should be updated to use the same yellow, like so:

![image](https://user-images.githubusercontent.com/355079/93216841-218c8600-f760-11ea-9c7a-00a482a581ce.png)

tombye commented:
I've pushed a few commits that:
- make the underline of the org links an equal thickness
- return the hover effect to the big numbers and the separate pill items

I think the big number hover wasn't happening before. At least it doesn't on preview at the moment, so I made a guess at how to put it back.

tombye commented:
I've done the following which, I believe, makes this ready for reviewing:
1. fixed the edit links on user list items (missed in previous work)
2. rebased all fixes for the same links/buttons into single commits

Number 1. turned this:
![image](https://user-images.githubusercontent.com/87140/93237644-4bea3d80-f778-11ea-9908-fa226dabcfea.png)
...into this:
![image](https://user-images.githubusercontent.com/87140/93237885-9ff52200-f778-11ea-9ef0-c80f1f11d077.png)

quis commented:
I found one place where the old yellow is still hiding:
![image](https://user-images.githubusercontent.com/355079/93338322-20bc2880-f822-11ea-916a-85a52e4ff6df.png)

For extra credit you could use [the monochrome version](https://github.com/alphagov/notifications-admin/blob/master/app/assets/images/folder-black-bold.svg) of the folder icon here:
![image](https://user-images.githubusercontent.com/355079/93338459-43e6d800-f822-11ea-8547-777e8e529a86.png)

Everything else looks excellent 👏

quis reviewed (approved):
I have weirdly enjoyed getting all these pixels lining up good. Thanks for all your work on this.
